### DISCUSSION TITLE
.aab from store crash on startup

### DISCUSSION THREAD

**Discussion by Coinlogru** [airsdk/Adobe-Runtime-Support#2036] [Q&A]
Hello,
Here is app https://play.google.com/store/apps/details?id=air.windjammerpoker

when run apk all work correct, but from aab a crash on startup (

adt.cfg contain path:
JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk-17.0.2.jdk/Contents/Home/
<img width="987" alt="image" src="https://user-images.githubusercontent.com/110308598/181995230-0371a24c-be30-4bde-b520-3db8dbb1cf58.png">

terminal: java -version
java version "17.0.2" 2022-01-18 LTS
Java(TM) SE Runtime Environment (build 17.0.2+8-LTS-86)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.2+8-LTS-86, mixed mode, sharing)

app contain Admob.ane
<img width="657" alt="image" src="https://user-images.githubusercontent.com/110308598/181995304-c7d554ac-ceac-4c28-8e6f-64b829f4ac0f.png">

Here is test logs from play console:
"at com.google.android.gms.internal.ads.zzaml.run(Unknown Source:4)"
[logcat (1).txt](https://github.com/airsdk/Adobe-Runtime-Support/files/9227536/logcat.1.txt)
[logcat.txt](https://github.com/airsdk/Adobe-Runtime-Support/files/9227576/logcat.txt)

where is error?

**Comment by marchbold** [comment]
Looks like you are missing the google play services extensions or at least the resources from those libraries:

```
07-29 05:34:17.372: E/GooglePlayServicesUtil(21799): The Google Play services resources were not found. Check your project configuration to ensure that the resources are included.
```

```
07-29 05:34:17.378: E/AndroidRuntime(21799): FATAL EXCEPTION: Thread-12
07-29 05:34:17.378: E/AndroidRuntime(21799): Process: air.windjammerpoker, PID: 21799
07-29 05:34:17.378: E/AndroidRuntime(21799): java.lang.NoClassDefFoundError: com.google.android.gms.common.R$string
```

Check that you are including the play services extensions in your app, not sure if you are using distriqt's or not, but they are available here: https://github.com/distriqt/ANE-GooglePlayServices/

**Reply by Coinlogru** [reply]
> Looks like you are missing the google play services extensions or at least the resources from those libraries:

> Check that you are including the play services extensions in your app, not sure if you are using distriqt's or not, but they are available here: https://github.com/distriqt/ANE-GooglePlayServices/

Thanks for answer! Not using distriqt, using https://github.com/pozirk/ANEAdMob but latest news of it is:
> 2022-03-29 pozirk: I'm not using or updating this ANE anymore. Have no idea, if it's still working.

But developer of it say, that google play services are in it's ane and say, that error could happen if aab compile inside Adobe Animate "box" as for only 32 of only for 64 bit?

> but they are available here: https://github.com/distriqt/ANE-GooglePlayServices/

Which one of them to import? For this game needed just Admob. And also is https://github.com/distriqt/ANE-GooglePlayServices/ will not conflict with https://github.com/pozirk/ANEAdMob ?

**Reply by marchbold** [reply]
If the developer of that ANE has packaged the Play Services libraries inside the ANE you won't be able to use ours as it will likely cause a conflict having 2 versions of the libs in your app. You could try to strip them out but that can be difficult. 

If you do manage to strip them out you can use our reference for Adverts to determine which of the support extensions are needed for AdMob: https://docs.airnativeextensions.com/docs/adverts/platform/admob

**Reply by Coinlogru** [reply]
Do you have an example or tutorial of how to add google ANE to load Admob rewarded ads in the Adobe Animate project with AIR for android app?
